velox/substrait/SubstraitToVeloxPlanValidator.cpp

```diff
@@ -195,6 +195,17 @@ bool SubstraitToVeloxPlanValidator::validate(
     return false;
   }
 
+  // In velox, the supported hash type in projectNode is 
+  // BOOLEAN, TINYINT, SMALLINT, INTEGER, BIGINT, VARCHAR, VARBINARY
+  // REAL, DOUBLE. Hash.cpp (L148 - L156)
+  for (auto i = 0; i < types.size(); i++) {
+    switch (types[i]->kind()) {
+      case TypeKind::DATE:
+        return false;
```

Comment:
I remember there is a case in TPC-H like `year(cast string_col as date)` which is supported by Velox, would this change bring fallback to that project?

Reply:
For case `year(cast string_col as date)` the project input type is string not date. So this change will not fallback that project.

Reply:
Makes sense to me. Please also notice there are somes fallback logics for unsupported type of specific function in Gluten, https://github.com/oap-project/gluten/blob/main/backends-velox/src/main/scala/io/glutenproject/utils/VeloxExpressionUtil.scala#L33-L42.
cc @lviiii